### ISSUE TITLE
Enforcement traffic signal: No warning

### ISSUE THREAD

**Issue by bitboy85** [osmandapp/OsmAnd#4291]
There is no warning if you reach an enforcement tagged as traffic sign, nether visual nor accoustic.
https://www.openstreetmap.org/edit#map=19/52.44048/7.07234

**Comment by hakuchi** [comment]
enforcement=traffic_signals([Red light camera](https://en.wikipedia.org/wiki/Red_light_camera)) should also give a warning.
I think this should also have a different message if I use TTS.
Something like: attention red light camera

**Comment by vshcherb** [comment]
enforcement traffic_sign are not supported yet.

**Comment by OpenSourceForEver** [comment]
Is there anything new about this? It would be very nice if there were anouncements of red light cameras.

**Comment by fansanelli** [comment]
+1

**Comment by cicku** [comment]
+1

**Comment by Duglof** [comment]
The least bad solution is to tag a speed camera instead of a red light camera to get an alert ...

**Comment by fansanelli** [comment]
Dear @Duglof 

this is a bad idea for multiple reasons:
- from the point of view of OSM we don't map for the rendering and even less for the routing... I wouldn't poison OSM data to achieve a warning in the navigation software;
- from the user experience perspective, I would expect a warning "slow down" (and maybe repeated every time my speed exceed just before and after the camera location), while for a "vista red" traffic light the warning should be "wait the green light" or something along this line. 
- last but not least, the speed camera is another beast and currently I think is outlaw in certain countries... By example in France you cannot have "safety cameras" position on the nav or you risk up to 1.500 E of fine, confiscation of the device and of the vehicle... To be extra-safe would be better to have a separate package for them (like wikipedia, tts, ...) especially for such regions.

Francesco

**Comment by Duglof** [comment]
The legal issue is already addressed, there is an option to remove speed cameras. 
The real question is why are the speed alerts are implemented and not the red light cameras ?
And don't you think people are tempted to address them differently in openstreetmap in order to get an alert ?

The good solution is that this alert is implemented in OsmAnd but it is not the case which forces some to make workarounds ...

**Comment by fansanelli** [comment]
I didn't want to go out topic, but safety camera are there, hidden under a flag... I think would be safer a separate package in this way the information on your phone and cannot get confiscated.
Anyway, I think the hard part of this task is that the red light camera are implemented with a relation, while speed cameras (usually) are just a node on the way object.
One workaround for users, could be to create a gpx file with overpass-turbo that contains the enforcements and add it as layer during the navigation.
Francesco

**Comment by Duglof** [comment]
I dont agree with you, speed cameras are also implemented with relations. 
- One-way speed cameras must be implemented with a relation in order to define the direction of control and give the allert only in One-Way.
- Some speed cameras are two-way and have a different speed limit depending on the direction of travel. ex 80km / h when North to South and 90 km/h when South to North (speed limit are declare inside the relation) !!

**Comment by fansanelli** [comment]
Are you sure that osmand support speed cameras as relation?

Francesco

**Comment by kozboko** [comment]
> Are you sure that osmand support speed cameras as relation?
> 
> Francesco

Yes, but they have generic POI overlay icons with a taxi (whyever). Addressed  here: https://github.com/osmandapp/OsmAnd-iOS/issues/2155

**Comment by hakuchi** [comment]
@vshcherb can you add this to the map creator, as you added the maxspeed cameras in here https://github.com/osmandapp/OsmAnd-tools/blob/3804b5f9aaaeba7a1f03e6f13799016bf1aae298/java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/preparation/IndexRouteCreator.java#L175 ?

**Comment by vshcherb** [comment]
Needs to be tested as already implemented - https://www.openstreetmap.org/relation/6932249#map=19/52.440750/7.073096

**Comment by Zirochkabila** [comment]
# Testing 1

> OsmAnd Maps 5324, released 23.01.26
> OsmAnd~ 5.3.0#7010m, Released: 2026-01-23

No audio warnings (or alerts) (iOS has a similar result)

<details><summary>Details</summary>
<video src="https://github.com/user-attachments/assets/c4ca9862-3c0b-427c-834f-9e61787aad72"></video>
</details>

**Comment by vshcherb** [comment]
Yes, we need to add support 
```
SPEED_CAMERA(1, R.string.traffic_warning_speed_camera),
	SPEED_LIMIT(2, R.string.traffic_warning_speed_limit),
	BORDER_CONTROL(3, R.string.traffic_warning_border_control),
	RAILWAY(4, R.string.traffic_warning_railways),
	TRAFFIC_CALMING(5, R.string.traffic_warning_calming),
	TOLL_BOOTH(6, R.string.traffic_warning_payment),
	STOP(7, R.string.traffic_warning_stop),
	PEDESTRIAN(8, R.string.traffic_warning_pedestrian),
	HAZARD(9, R.string.traffic_warning_hazard),
	MAXIMUM(10, R.string.traffic_warning),
	TUNNEL(11, R.string.tunnel_warning);
```

**Comment by Zirochkabila** [comment]
# Testing 2

> OsmAnd Maps 5447, released 19.08.26
> OsmAnd~ 5.4.0#8575m, released: 2026-08-19

- [ ] We have notifications for “Red light camera,” but there are still none for “traffic_signals”






<details><summary>Details</summary>
<table style="width:100%">
  <tr>
    <td><video source src="https://github.com/user-attachments/assets/61e06067-e5ef-4b12-a947-ccffdd2dc95c"  width="300" height="auto"/></td>
    <td><video source src="https://github.com/user-attachments/assets/5b796b0a-a1fc-481c-b9d3-ca2d5ef4d5cc"  width="300" height="auto"/> </td>
  </tr>
</table> 
</details>

**Comment by vshcherb** [comment]
We will fix as separate issue if requested.

**Comment by Zirochkabila** [comment]
@vshcherb that's exactly what was in @bitboy85 initial request

> There is no warning if you reach an enforcement tagged as traffic sign, nether visual nor accoustic.
> https://www.openstreetmap.org/edit#map=19/52.44048/7.07234
